add

MaximGolyak opened the pull request.

MaximGolyak committed:
add

mateacademy-ai-mentor reviewed (changes requested):
Hi! 👋 The `MainApp.java` file is missing the core implementation. You need to create the `Machine`, `Truck`, `Bulldozer`, and `Excavator` classes, then create an array of `Machine` objects with instances of `Truck`, `Bulldozer`, and `Excavator`. Finally, iterate through this array, calling the `doWork()` and `stopWork()` methods on each object. Please add this logic to complete the task. 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

🤖 If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) field below and **mention me with `@mateacademy-ai-mentor` tag**. I will read your message when I perform the next review of your pull request.

MaximGolyak committed:
added class Excavator
added class Bulldozer
added class Truck
added abstract class Machine
added public abstract void doWork() and
      public abstract void stopWork();

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **Something went wrong while processing the review.**
> <sub>*[CODE: 3]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **Something went wrong while processing the review.**
> <sub>*[CODE: 3]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

The code mostly meets the requirements! 🎉 However, let's make those `doWork()` messages a bit more informative! Currently, they're a little vague, like just saying "Working...". Instead, let's specify *which* machine is doing the work. For example, instead of a generic "Working...", a `Bulldozer`'s `doWork()` could print "Bulldozer is working!". Similarly, a `Truck`'s `doWork()` could print "Truck is working!". This will make the output much clearer and easier to understand when you have multiple machines working together. This is important for debugging and understanding the flow of execution in your program. Keep up the great work, and remember to review the comments to improve the messages.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
